File: EuDecorator/Controllers/Dtos/ValidationInitializeResponse.cs
using System.Text.Json.Serialization;

namespace EuDecorator.Controllers.Dtos
{

    /// <summary>
    /// 3.8.3
    /// The validation initialisation response delivers a JSON with a unique subject ID, which identifies the
    /// occurrence in the validation service.Next to the unique subject, a public key and an expiration date
    /// for the occurrence are attached. The public key must be unique for each subject for maximum data
    /// privacy.
    /// </summary>
    public class ValidationInitializeResponse
    {
        /// <summary>
        /// hexadecimal-encoded value
        /// Traveller or whole trip?
        /// </summary>
        [JsonPropertyName("sub")]
        public string SubjectId { get; set; }

        /// <summary>
        /// Unix epoch time
        /// </summary>
        [JsonPropertyName("exp")]
        public int WhenExpires { get; set; }

        /// <summary>
        /// E.g. https://validationprovider/validate/{subject}
        /// TODO odd name
        /// </summary>
        [JsonPropertyName("aud")]
        public int ValidationUrl { get; set; }
        
        /// <summary>
        /// "Optional Public Key for Encryption of Validation Service."
        /// NOTE: This key should only be used in a Backend-Backend Validation.
        /// A wallet should not use the given key provided by the service provider backend, because this one can be intercepted.
        /// 
        /// Hence this is NOT the key to use to encrypt DDCs...
        /// </summary>
        [JsonPropertyName("encKey")]
        public JsonWebKeyRfc7517 ValidationServiceEncryptionKey { get; set; }

        /// <summary>
        /// "Optional Public Key for Signing of Validation Service."
        /// See encKey...
        /// </summary>
        [JsonPropertyName("signKey")]
        public JsonWebKeyRfc7517 ValidationServiceSigningKey { get; set; }
    }
}

File: src/main/kotlin/nl/rijksoverheid/minienw/travelvalidation/validationservice/api/data/initialize/ValidationInitializeResponse.kt
﻿package nl.rijksoverheid.minienw.travelvalidation.validationservice.api.data.initialize

import com.fasterxml.jackson.annotation.JsonProperty
import nl.rijksoverheid.minienw.travelvalidation.validationservice.api.data.PublicKeyJwk

/**
    * 3.8.3
    * The validation initialisation response delivers a JSON with a unique subject ID, which identifies the
    * occurrence in the validation validationservice.Next to the unique subject, a var key and an expiration date
    * for the occurrence are attached.The var key must be unique for each subject for maximum data
    * privacy.
    */
class ValidationInitializeResponse(
    /**
     * hexadecimal-encoded value
     * Traveller or whole trip?
     */
    @JsonProperty("sub")
    var subjectId: String,

    /**
     * Unix epoch time
     */
    @JsonProperty("exp")
    var whenExpires: Long,

    /**
     * E.g. https://validationprovider/validate/{subject}
     * TODO odd name
     * Moved to Identity end point... :(
     */
    //@JsonProperty("aud")
    //var ValidationUrl: Int,

    /**
     * "Optional var Key for Encryption of Validation nl.rijksoverheid.minienw.travelvalidation.validationservice.api.data.identity.Service."
     * TODO should be using RSA not AES as an AES secret key should be exchange with DiffieHellman first.
     */
    @JsonProperty("encKey")
    var validationServiceEncryptionKey: PublicKeyJwk?,

    /**
     * "Optional var Key for Signing of Validation nl.rijksoverheid.minienw.travelvalidation.validationservice.api.data.identity.Service."
     * See encKey...
     * TODO NB why bother as the wallet has already sent you a public key and GET/identity provides the one for the validationservice
     */
    @JsonProperty("signKey")
    var signKey: PublicKeyJwk?,
)